.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -64,29 +64,53 @@ jobs:
       - name: Checkout Source code
         uses: actions/checkout@v2
 
-      - name: Install Rust Toolchain
-        uses: actions-rs/toolchain@v1
-
-      - name: Install WASM target
-        run: rustup target add wasm32-unknown-unknown
-
       - name: Download artifact
         uses: actions/download-artifact@v2
         with:
           name: aleph-release-runtime
 
-      - name: Build runtime and compare checksum with artifact
+      - name: Compute checksum of aleph-runtime
         env:
           ARTIFACT: aleph_runtime.compact.wasm
           TARGET_DIR: target/release/wbuild/aleph-runtime
         run: |
           mkdir -p "$TARGET_DIR"
           mv "$ARTIFACT" "$TARGET_DIR"
           sha256sum "$TARGET_DIR/$ARTIFACT" > checksum.sha256
-          cargo clean
-          cargo build --release -p aleph-runtime
-          sha256sum -c checksum.sha256
+          rm -rf "$TARGET_DIR"
+
+      - name: Checkout docker image
+        uses: actions/cache@v2
+        with:
+          path: |
+            image.tar
+          key: ${{ runner.os }}-build-docker-image-${{ hashFiles('**/Cargo.lock', '**/rust-toolchain', 'nix/nix-build.sh') }}-v0
```

Comment:
I wonder whether it makes sense to extract additional workflow (with trigger `on workflow_call` etc) , and call it from this one and `build-node-and-runtime.yml`, in order to decrease code duplication. Things we call twice now is e.g. `hashFiles('**/Cargo.lock', '**/rust-toolchain', 'nix/nix-build.sh')` which is a concrete logic that might change in the future, so better to have it in one place. Also, there's some boilerplate code around calling nix with docker, so I imagine this new workflow would have arguments like
* `crates` 
* `single_step`
And would define those steps
* "Build or load our `build` docker image"
* "Build package"
* ... etc

Reply:
Moved to a separate action.